homeassistant/components/bsblan/config_flow.py

```diff
@@ -41,6 +44,68 @@ async def async_step_user(
         self.username = user_input.get(CONF_USERNAME)
         self.password = user_input.get(CONF_PASSWORD)
 
+        return await self._validate_and_create()
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle Zeroconf discovery."""
+
+        self.host = str(discovery_info.ip_address)
+        self.port = discovery_info.port or DEFAULT_PORT
+
+        # Try to get MAC from properties first (for future compatibility)
+        self.mac = discovery_info.properties.get("mac")
+
+        # If MAC not found in properties, try to parse from raw TXT records
+        if not self.mac and hasattr(discovery_info, "properties_raw"):
+            for key in discovery_info.properties_raw:
+                if key.startswith(b"mac="):
+                    self.mac = key[4:].decode("utf-8")
+                    break
+
+        # If still no MAC found, abort the discovery flow
+        if not self.mac:
+            return self.async_abort(reason="no_mac_address")
```

Comment:
Isn't the Mac always present?

Reply:
Used _async_handle_discovery_without_unique_id() for discovery without MAC identifier.